org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/IRenameNameValidator.xtend

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.refactoring
+
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.xtext.conversion.IValueConverterService
+import org.eclipse.xtext.conversion.ValueConverterException
+
+import static org.eclipse.xtext.ide.refactoring.RefactoringIssueAcceptor.Severity.*
+import static org.eclipse.xtext.util.Strings.equal
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+@ImplementedBy(RuleBasedNameValidator)
+interface IRenameNameValidator {
+
+	def void validate(EObject target, String newName, RefactoringIssueAcceptor issues)
+
+	class RuleBasedNameValidator implements IRenameNameValidator {
+
+		@Inject IValueConverterService valueConverterService
+
+		override validate(EObject target, String newName, RefactoringIssueAcceptor issues) {
+			try {
+				var String value = getNameAsValue(newName)
+				var String text = getNameAsText(value)
+				if (!equal(text, newName)) {
+					issues.add(ERROR, '''Illegal name: '�newName�'. Consider using '�text�' instead.''')
+				}
```

Comment:
In case a value converter normalizes values this validation enforces that users specify the already-normalized value. 
For example, for the following value converter users would not be allowed to use "null" as a value even though during parsing it would be allowed.
```
int getValue(String it) {
  case "0": return 0;
  case "null": return 0;
}
String getText(int it) {
  return String.valueOf(it)
}
```